autogen/agentchat/contrib/graph_rag/graph_rag_agent.py

```diff
@@ -0,0 +1,44 @@
+from abc import ABC, abstractmethod
+from typing import List
+
+from autogen.agentchat import ConversableAgent
+
+from .graph_store import GraphStore
+
+
+class GraphRagAgent(ConversableAgent, ABC):
```

Comment:
What about using `Protocol` instead of `ABC` https://github.com/microsoft/autogen/blob/main/autogen/agentchat/agent.py ?